mozilla/foundation.mozilla.org
# Filter bar for the directory component

Related PRs/issues #2669 #2685

**Changes in Models:**
There is a migration in this PR, but that migration is already covered by #2669 and will disappear from this PR on a rebase.

**Documentation:**

**Commits**
This builds on another, not yet merged, PR, so most of the "files changed" are not actually part of this PR. The relevant changes are specifically in https://github.com/mozilla/foundation.mozilla.org/pull/2698/commits/0286cd09fb68dfb5cc7272d2580f964bbdd4e479

**How to test**
short version: https://foundation-mofostaging-pr-2698.herokuapp.com/cms/pages/21/edit/ for the CMS side, https://foundation-mofostaging-pr-2698.herokuapp.com/en/campaigns/single-page/ for the example

long version: check out the branch, and...
- add `ENABLE_PROFILE_DIRECTORY_BLOCK=True` to your `.env` file (hurray, feature flags)
- run `inv catch-up` (just to make sure. This also runs the migrations)
- if you started on an empty database, `inv manage load_fake_data` is a good idea, so you have something to edit
- run `npm start` to start everything up and then once running head over the `http://localhost:8000/cms" and log in as usual (if you had an empty db, you might need to run the createsuperuser command)
- head over to the single-page campaign page, and edit it: add a `profile directory` component to the streamfield body, and set the profile type to `fellow`, and the year filter to `2013,2014,2015,2016,2017,2018,2019`, then "save draft" (or publish) and then view the draft (or live page).

Scroll to the profile listing, and hit the various filter buttons to see content getting swapped in based on the year you picked.
